cmd/frcli/node_report.go

```diff
@@ -0,0 +1,100 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path"
+	"strings"
+	"time"
+
+	"github.com/lightninglabs/faraday/frdrpc"
+	"github.com/urfave/cli"
+)
+
+var onChainReportCommand = cli.Command{
+	Name:     "nodereport",
+	Category: "reporting",
+	Usage:    "Get a report of node activity.",
+	Flags: []cli.Flag{
+		cli.Int64Flag{
+			Name: "start_time",
+			Usage: "(optional) The unix timestamp in seconds " +
+				"from which the report should be generated," +
+				"defaults to one week ago",
+		},
+		cli.Int64Flag{
+			Name: "end_time",
+			Usage: "(optional) The unix timestamp in seconds" +
+				"until which the report should be generated." +
+				"If not set, the report will be produced " +
+				"until the present.",
+		},
+		cli.StringFlag{
+			Name: "csv_path",
+			Usage: "A path to write node_report.csv to. If not " +
+				"set, the command will output the report. " +
+				"Note that write permissions are required.",
+		},
+	},
+	Action: queryOnChainReport,
+}
+
+func queryOnChainReport(ctx *cli.Context) error {
+	client, cleanup := getClient(ctx)
+	defer cleanup()
+
+	// Set start and end times from user specified values, defaulting
+	// to zero if they are not set.
+	req := &frdrpc.NodeReportRequest{
+		StartTime:   uint64(ctx.Int64("start_time")),
+		EndTime:     uint64(ctx.Int64("end_time")),
+		Granularity: frdrpc.FiatEstimateRequest_HOUR,
+	}
+
+	// If start time is zero, default to a week ago.
+	if req.StartTime == 0 {
+		weekAgo := time.Now().Add(time.Hour * 24 * 7 * -1)
+		req.StartTime = uint64(weekAgo.Unix())
+	}
+
+	// If end time is zero, default to the present.
+	if req.EndTime == 0 {
+		req.EndTime = uint64(time.Now().Unix())
```

Comment:
nit: this is not needed, as i think this case is also handled on the server side.